pkg/client/vcwallet/client.go

```diff
@@ -225,12 +225,17 @@ func (c *Client) Issue(credential json.RawMessage,
 // Prove produces a Verifiable Presentation.
 //
 //	Args:
-//		- List of verifiable credentials IDs.
-//		- Proof options
+//		- list of interfaces (string of credential IDs which can be resolvable to stored credentials in wallet or
+//		raw credential).
+//		- proof options
 //
-func (c *Client) Prove(credentialIDs []string, options *wallet.ProofOptions) (json.RawMessage, error) {
-	// TODO to be added #2433
-	return nil, fmt.Errorf("to be implemented")
+func (c *Client) Prove(credentials []interface{}, options *wallet.ProofOptions) (*verifiable.Presentation, error) {
```

Comment:
I'd rather not use an interface[]. Looking at the code, you have two different inputs to Prove: credential IDs or credentials.

Reply:
Also - It seems you switched from Prove being on stored credentials to being on anything passed-in. What was the reason?

Reply:
If you really need different inputs into the same function, you could also look at how NewPresentation was done:
https://github.com/hyperledger/aries-framework-go/blob/main/pkg/doc/verifiable/presentation.go#L179-L221

Reply:
@troyronda things would have been simpler ([]string) if it is just for stored VCs. During selective disclosure, derived VCs aren't saved and we can not generate presentation unless we save it. But we cannot save them since it might replace the original VC (due to same `id` property)

Reply:
fixed, added variadic args for passing different kind of credential arguments.